via/services/checkmate.py

```diff
@@ -5,13 +5,20 @@
 from via.exceptions import BadURL
 
 
-class ViaCheckmateClient(CheckmateClient):
-    def __init__(self, request):
-        super().__init__(
-            host=request.registry.settings["checkmate_url"],
-            api_key=request.registry.settings["checkmate_api_key"],
+class CheckmateService:
+    def __init__(self, checkmate_client, allow_all, blocked_for, ignore_reasons):
+        self._checkmate_client = checkmate_client
+        self._allow_all = allow_all
+        self._blocked_for = blocked_for
+        self._ignore_reasons = ignore_reasons
+
+    def check_url(self, url):
+        return self._checkmate_client.check_url(
+            url,
+            allow_all=self._allow_all,
+            blocked_for=self._blocked_for,
+            ignore_reasons=self._ignore_reasons,
```

Comment:
Because we're using composition `ViaClientService` has to have an explicit `check_url()` method that forwards to `checkmatelib`'s `CheckmateClient.check_url()`.

As usual with composition this can seem a bit wordier (having to have this boilerplate forwarding method and tests for it) but it's much clearer: you can see what the API of the `CheckmateService` class it by just looking at the code of the class itself, without also having to look at the `CheckmateClient` class (which is in an external library), and no confusion about calls to a `check_url()` method that doesn't seem to be defined anywhere.

This `check_url()` method also isn't *just* forwarding to `CheckmateClient.check_url()`: it's also injecting the `allow_all`, `blocked_for` and `ignore_reasons` methods. This means that `raise_if_blocked()` (below) doesn't need to inject those and makes that method shorter and clearer by separating the concerns.